common/src/lib.rs

```diff
@@ -2,15 +2,79 @@ pub mod crypto;
 
 use crypto::*;
 use serde::{Deserialize, Serialize};
-use std::collections::BTreeSet;
+use std::collections::{BTreeMap, BTreeSet};
 use thiserror::Error;
 
+/// A set of state that is directly recorded in the header.
+///
+/// This state effects the consensus, unlike the ordinary state (which is used for data recording).
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub struct EssentialState {
+    /// The original validator set information before the delegation calculation.
+    ///
+    /// The order here is the priority order of the validators
```

Comment:
priority order in voting power?

Reply:
no voting power, leader priority. added some extra explanation.